base_layer/wallet/src/key_manager_service/storage/sqlite_db/key_manager_state.rs

```diff
@@ -0,0 +1,225 @@
+                km.primary_key_index = new_index.to_le_bytes().to_vec();
+                if let Some(cip) = cipher {
+                    km.encrypt(cip).map_err(KeyManagerStorageError::AeadError)?;
+                }
+                let update = KeyManagerStateUpdateSql {
+                    branch_seed: None,
+                    primary_key_index: Some(km.primary_key_index),
+                };
+                diesel::update(key_manager_states::table.filter(key_manager_states::id.eq(&km.id)))
+                    .set(update)
+                    .execute(conn)
+                    .num_rows_affected_or_not_found(1)?;
+                new_index
+            },
+            Err(_) => return Err(KeyManagerStorageError::KeyManagerNotInitialized),
+        })
+    }
+
+    pub fn set_index(
+        branch: String,
+        index: u64,
+        cipher: Option<&Aes256Gcm>,
+        conn: &SqliteConnection,
+    ) -> Result<(), KeyManagerStorageError> {
+        match KeyManagerStateSql::get_state(&branch, conn) {
+            Ok(mut km) => {
+                km.primary_key_index = index.to_le_bytes().to_vec();
+                if let Some(cip) = cipher {
+                    km.encrypt(cip).map_err(KeyManagerStorageError::AeadError)?;
+                }
+                let update = KeyManagerStateUpdateSql {
+                    branch_seed: None,
+                    primary_key_index: Some(km.primary_key_index),
+                };
+                diesel::update(key_manager_states::table.filter(key_manager_states::id.eq(&km.id)))
+                    .set(update)
+                    .execute(conn)
+                    .num_rows_affected_or_not_found(1)?;
+                Ok(())
+            },
+            Err(_) => Err(KeyManagerStorageError::KeyManagerNotInitialized),
+        }
+    }
+}
+
+#[derive(AsChangeset)]
+#[table_name = "key_manager_states"]
+pub struct KeyManagerStateUpdateSql {
+    branch_seed: Option<String>,
+    primary_key_index: Option<Vec<u8>>,
+}
+
+impl Encryptable<Aes256Gcm> for KeyManagerStateSql {
+    fn encrypt(&mut self, cipher: &Aes256Gcm) -> Result<(), String> {
+        let encrypted_index = encrypt_bytes_integral_nonce(cipher, self.primary_key_index.clone())?;
+        self.primary_key_index = encrypted_index;
+        Ok(())
+    }
+
+    fn decrypt(&mut self, cipher: &Aes256Gcm) -> Result<(), String> {
+        let decrypted_index = decrypt_bytes_integral_nonce(cipher, self.primary_key_index.clone())?;
+        self.primary_key_index = decrypted_index;
+
+        Ok(())
+    }
+}
+
+impl Encryptable<Aes256Gcm> for NewKeyManagerStateSql {
+    fn encrypt(&mut self, cipher: &Aes256Gcm) -> Result<(), String> {
+        let encrypted_index = encrypt_bytes_integral_nonce(cipher, self.primary_key_index.clone())?;
+        self.primary_key_index = encrypted_index;
+        Ok(())
+    }
+
+    fn decrypt(&mut self, _cipher: &Aes256Gcm) -> Result<(), String> {
+        unimplemented!("Not supported")
```

Comment:
I guess rather just return this as an `Err` instead of panicing

Reply:
This should panic. You are never supposed to be able to decrypt a new_key_manager_state. 
It should not be possible. new_key_manager_state is only used when inserting a new key_manager_state into the DB, and there after if you retrieve it you will use key_manager_state.